dali/pipeline/data/meta.h

```diff
@@ -40,10 +40,14 @@ class DALIMeta {
     return source_info_;
   }
 
-  inline void SetSourceInfo(std::string source_info) {
+  inline void SetSourceInfo(const std::string &source_info) {
     source_info_ = source_info;
   }
 
+  inline void SetSourceInfo(std::string &&source_info) {
+    source_info_ = std::move(source_info);
```

Comment:
Why not having std::forward here and letting it determine the value category of the arg?

Reply:
Exactly.

Reply:
Oh right